daml/Marketplace/Distribution/Auction/Service.daml

```diff
@@ -0,0 +1,201 @@
+{-# LANGUAGE MultiWayIf #-}
+
+module Marketplace.Distribution.Auction.Service where
+
+import DA.Action (foldlA)
+import DA.Foldable (forA_)
+import DA.List (sortOn, mapAccumL)
+import DA.Finance.Asset (AssetDeposit)
+import DA.Finance.Types (Account, Asset, Id)
+import DA.Finance.Utils (fetchAndArchive)
+import Marketplace.Settlement qualified as Settlement
+import Marketplace.Distribution.Auction.Model qualified as Auction
+import Marketplace.Distribution.Auction.Model (Auction(..))
+import Marketplace.Distribution.Auction.Utils (dutchAuction, splitList, withdrawAllocation, updateBidStatus, generateSettlementInstruction)
+import Marketplace.Distribution.Bid.Service qualified as Bid
+import Marketplace.Distribution.Bid.Model qualified as Bid
+import Marketplace.Trading.AllocationAccountRule qualified as Allocation
+import Marketplace.Trading.AllocationAccountRule (AllocationAccountRule(..))
+
+type S = Service
+
+template Service
+  with
+    operator : Party
+    provider : Party
+    customer : Party
+  where
+    signatory operator, provider, customer
+
+    nonconsuming choice OfferBidder : ContractId Bid.Offer
+      with
+        submitter : Party
+        auctionCid : ContractId Auction.T
+        bidder : Party
+      controller submitter
+      do
+        assertMsg ("Unauthorised Submitter - " <> show submitter) (submitter `elem` [provider, customer])
```

Comment:
I don't quite get why this is necessary.
Is it possible to instead have controller-first `provider, customer` choice statement? Wouldn't that solve this problem?

Reply:
in fact I think it should be only the provider that's able to offer bidding services to the bidders

Reply:
The reason for its current implementation was that I presumed either a `provider` or a `customer` should be able to call these choices. If I have a controller with `provider, customer can`, then it means the caller of the choice must have both signatures in order to exercise that choice - this pattern avoids that authorisation requirement but still allows either party to call it. 

As per @GeorgSchneider 's comment, if only the `provider` can call those choices then they can be easily refactored to reflect that.